scripts/docker/Dockerfile.clang

```diff
@@ -1,4 +1,4 @@
-FROM nvidia/cuda:9.2-devel
+FROM nvidia/cuda:12-devel
```

Comment:
The naming scheme has changed. See https://catalog.ngc.nvidia.com/orgs/nvidia/containers/cuda/tags for the possible tags